source/features/feature-descriptions.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import domify from 'doma';
+import elementReady from 'element-ready';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {wrapAll} from '../helpers/dom-utils';
+import GitHubURL from '../github-helpers/github-url';
+import parseBackticks from '../github-helpers/parse-backticks';
+
+interface Commit {
+	messageHeadline: string;
+	messageHeadlineHTML: string;
+	commitUrl: string;
+	committedDate: string;
+}
+
+const getFeatureHistory = cache.function(async (fileName: string): Promise<Commit[]> => {
+	const {repository} = await api.v4(`
+		repository() {
+			defaultBranchRef {
+				target {
+					...on Commit {
+						history(first:100, path: "source/features/${fileName}") {
+							nodes {
+								messageHeadline
+								messageHeadlineHTML
+								commitUrl
+								committedDate
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+	return repository.defaultBranchRef.target.history.nodes;
+}, {
+	maxAge: {days: 1},
+	staleWhileRevalidate: {days: 4},
+	cacheKey: ([fileName]): string => __filebasename + ':' + fileName
+});
+
+function getCommitUrl(commit: Commit): string {
+	const [, pullRequestUrl] = /<a[^>]+href="([^"]+)">/.exec(commit.messageHeadlineHTML) ?? [];
+	return pullRequestUrl ?? commit.commitUrl;
+}
+
+async function getHistoryDropdown(featureName: string): Promise<Element> {
+	let history = await getFeatureHistory(featureName + '.tsx');
+
+	if (history.length === 0) {
+		// Feature might be CSS-only
+		history = await getFeatureHistory(featureName + '.css');
+	}
+
+	const filteredHistory = history.filter((commit: Commit) => !/^Meta|^Document|^Readme|^Lint|^Update.+dependencies/.test(commit.messageHeadline));
+	const skippedCommitsCount = history.length - filteredHistory.length;
+
+	const historyUrl = new GitHubURL(location.href);
+	historyUrl.assign({route: 'commits'});
+
+	return (
+		<details className="dropdown details-reset details-overlay d-inline-block">
+			<summary className="text-gray d-inline" aria-haspopup="true">
+				Feature history
+				<div className="dropdown-caret ml-1"/>
+			</summary>
+
+			<div className="dropdown-menu dropdown-menu-s" style={{width: 400}}>
+				<ul className="overflow-y-auto" style={{maxHeight: '60vh'}}>
```

Comment:
Is the height necessary? The whole page can scroll instead

Reply:
It doesn't need to, but it looks weird on very small features with a long history, where the dropdown menu is longer than the page itself.